spock-core/src/main/java/org/spockframework/mock/runtime/DynamicProxyMockInterceptorAdapter.java

```diff
@@ -25,6 +27,34 @@ public DynamicProxyMockInterceptorAdapter(IProxyBasedMockInterceptor interceptor
   }
 
   public Object invoke(Object target, Method method, Object[] arguments) throws Throwable {
+    boolean isDefault = (method.getModifiers() & (Modifier.ABSTRACT | Modifier.PUBLIC | Modifier.STATIC)) == Modifier.PUBLIC
```

Comment:
This expression is wrong
`(method.getModifiers() & (Modifier.ABSTRACT | Modifier.PUBLIC | Modifier.STATIC)) == Modifier.PUBLIC` it is only true if the modifier is PUBLIC. You probably meant 
```
int publicAbstractStatic = (Modifier.ABSTRACT | Modifier.PUBLIC | Modifier.STATIC);
(method.getModifiers() & publicAbstractStatic ) == publicAbstractStatic
```